### PR TITLE
Update vulnerabilities & add readme

### PR DESCRIPTION
**Pull request by pxieTG**
- decouple jdbc from spark-core(get truststore files on Spark side)
  - uber jar size from 100MB to 1.6MB
  - eliminate conflicts from spark, hdfs, scala, etc.
  - remove log4j
- Update some vulnerabilities(**test passed**)
  - ~~Update log4j to 2.17.2~~
  - Update httpclient to 4.5.13
- Separate profiles for publishing(**test passed**)
  Add a profile `publish`, `sign` and `deploy` can only be triggered in this profile
- Add readme section "Download from Maven Central Repository"

### CODE REVIEW

**Review by chengjie-qin — APPROVED** [review]

File: tools/etl/tg-jdbc-driver/README.md [hunk 1]
```diff
@@ -8,18 +8,52 @@ The TigerGraph JDBC Driver is a Type 4 driver, converting JDBC calls directly in
...
+| org.junit.jupiter         | junit-jupiter-engine | 5.8.2 |
+| org.junit.vintage | junit-vintage-engine | 5.8.2 |
+| org.apache.logging.log4j | log4j-core | 2.17.2 |
+| org.apache.spark | spark-core_2.13 | 3.2.0 |
+
+## Download from Maven Central Repository
+
+Tigergraph JDBC Driver can be searched in https://search.maven.org as `tigergraph-jdbc-driver`.
+
+Please refer to [Versions compatibility](#Versions compatibility) to find the **suitable version**.
```
Comment (chengjie-qin) [inline tools/etl/tg-jdbc-driver/README.md:29]: The link to `Versions compatibility` is not working. You can see from here: https://github.com/pxieTG/ecosys/tree/master/tools/etl/tg-jdbc-driver

File: tools/etl/tg-jdbc-driver/README.md [hunk 2]
```diff
@@ -8,18 +8,52 @@ The TigerGraph JDBC Driver is a Type 4 driver, converting JDBC calls directly in
 | --- | --- | --- | --- | --- | --- |
 | 1.0 | 2.2.4+ | 1.8 | Rest++ | JSON | Support builtin, compiled queries and loading jobs |
 | 1.1 | 2.4.1+ | 1.8 | Rest++ | ResultSet | Support tabular format and Spark |
-| 1.2 | 2.4.1+ | 1.8 | Rest++ | ResultSet | Support interpreted queries and Spark partitioning |
+| 1.2.x | 2.4.1~3.4 | 1.8 | Rest++ | ResultSet | Support interpreted queries and Spark partitioning |
```
Comment (TanYongSql) [inline tools/etl/tg-jdbc-driver/README.md:11]: There're just v1.2.1 and v1.2.2 on maven.com, shall we remove entries for v1.0 and v1.1 and add v1.2.1?

File: tools/etl/tg-jdbc-driver/README.md [hunk 3]
```diff
@@ -8,18 +8,52 @@ The TigerGraph JDBC Driver is a Type 4 driver, converting JDBC calls directly in
...
 | org.glassfish | javax.json | 1.1.4 |
-| junit | junit | 4.11 |
+| org.junit.jupiter         | junit-jupiter-engine | 5.8.2 |
+| org.junit.vintage | junit-vintage-engine | 5.8.2 |
+| org.apache.logging.log4j | log4j-core | 2.17.2 |
+| org.apache.spark | spark-core_2.13 | 3.2.0 |
+
+## Download from Maven Central Repository
+
+Tigergraph JDBC Driver can be searched in https://search.maven.org as `tigergraph-jdbc-driver`.
```
Comment (TanYongSql) [inline tools/etl/tg-jdbc-driver/README.md:27]: How about this?
Tigergraph JDBC Driver can be found at [maven.org](https://search.maven.org/artifact/com.tigergraph/tigergraph-jdbc-driver).

File: tools/etl/tg-jdbc-driver/README.md [hunk 4]
```diff
@@ -8,18 +8,52 @@ The TigerGraph JDBC Driver is a Type 4 driver, converting JDBC calls directly in
...
+
+    ```groovy
+    implementation 'com.tigergraph:tigergraph-jdbc-driver:{SUITABLE_VERSION}'
+    ```
+
+  Your build automation tool will download it autometically from maven central repository.
+
+* Use the jar file directly (e.g. Spark):
+
+  Search for `tigergraph-jdbc-driver` in https://search.maven.org to get the jar file.
```
Comment (TanYongSql) [inline tools/etl/tg-jdbc-driver/README.md:53]: How about this?
`Go to [maven.org](https://search.maven.org/artifact/com.tigergraph/tigergraph-jdbc-driver) to get the jar file.`

File: tools/etl/tg-jdbc-driver/README.md [hunk 5]
```diff
@@ -6,20 +6,53 @@ The TigerGraph JDBC Driver is a Type 4 driver, converting JDBC calls directly in
 
 | JDBC Version | TigerGraph Version | Java | Protocol | Query Result Format | New Features |
 | --- | --- | --- | --- | --- | --- |
-| 1.0 | 2.2.4+ | 1.8 | Rest++ | JSON | Support builtin, compiled queries and loading jobs |
-| 1.1 | 2.4.1+ | 1.8 | Rest++ | ResultSet | Support tabular format and Spark |
-| 1.2 | 2.4.1+ | 1.8 | Rest++ | ResultSet | Support interpreted queries and Spark partitioning |
+| 1.2.1 | 2.4.1~3.4 | 1.8 | Rest++ | ResultSet | Support interpreted queries and Spark partitioning |
+| 1.2.2 | 2.4.1~3.4 | 1.8 | Rest++ | ResultSet | Support interpreted queries and Spark partitioning |
```
Comment (TanYongSql) [inline tools/etl/tg-jdbc-driver/README.md:10]: For v1.2.2's new feature, you may just say bug fix.

**Review by TanYongSql — APPROVED** [review]

**Pull request merged**